value.go

```diff
@@ -1170,8 +1170,6 @@ func (o *Object) MemUsage(ctx *MemUsageContext) (uint64, error) {
 		return SizeEmpty, nil
 	case *objectGoMapReflect:
 		return SizeEmpty, nil
-	case *objectGoMapSimple:
-		return SizeEmpty, nil
```

Comment:
[major] The fact that no test is failing worries me as this is a "breaking" change. Can we add tests even just around this `MemUsage` function? You can look at `toValue` to see how it's used or leverage that directly in the test body.

I think this will require to add a `MemUsage` function to `objectGoMapSimple`. But in order to calculate it you would can call the `MemUsage` function from the embedded `baseObject` + you have to calculate it for the `data` field.

The catch here is that you have to loop through key/values and calculate the memory usage for each pair. Keys mem usage is calculated as `uint64(len(k)) + SizeString`, but `SizeString` is not available yet and it's part of my PR (you can just copy paste it here, it's a constant value anyway). For the values it's probably easier to convert them to a value (call ToValue) and then use the `MemUsage` on them (I think you can just call `o._getStr(k)` and use the return value to call `MemUsage` on it.

Let me know if you have any questions!

Reply:
Roger, Hmmm okay, yeah it appeared to be incrementing the fields when I ran this in the debugger against a VM test, but it also didn't fail the test. I'm now wondering if the MemUsage is already managing to take into account what's contained in the map... I'll write some tests as I investigate.

Reply:
I ran the debugger for the BAAS test `TestMemUsageBSON` found here. I noticed it looks like if we add this we will also start tracking the memory used by our `context`, `console`, etc... For example: 
![Screen Shot 2023-08-02 at 2 59 12 PM](https://github.com/mongodb-forks/goja/assets/7692062/974e7e82-b7ec-463b-949d-dd41f8499a0b)
![Screen Shot 2023-08-02 at 2 59 05 PM](https://github.com/mongodb-forks/goja/assets/7692062/cd190af7-07ef-4b7d-93c7-87afaf301c39)
![Screen Shot 2023-08-02 at 2 58 58 PM](https://github.com/mongodb-forks/goja/assets/7692062/24a15097-3d74-474e-9f01-013b1a34d051)
![Screen Shot 2023-08-02 at 2 58 51 PM](https://github.com/mongodb-forks/goja/assets/7692062/3c9bf8af-6f0c-4dbb-9365-6878c70195be)

Reply:
Wasn't sure if this would be an unexpected change and whether we want to capture it or not.

Reply:
Just wanted to confirm if you are correct, for my example test [here](https://github.com/mongodb-forks/goja/blob/13161c04bc3ac92c2a8728b8ac92078e5d91ac36/value_test.go#L61), it looks like the default memory usage implementation for this object is substantially larger than if we implemented it ourselves. [evg result](https://parsley.mongodb.com/evergreen/goja_linux_64_test_patch_96936aa462710bb3f9bd16e8b379cd5465592a42_64cac111d6d80a4b794d50a0_23_08_02_20_48_18/0/task?bookmarks=0,41215&shareLine=40923)

Reply:
The much higher usage is normal, if you look at some of tests in [memory_test.go you see that we do a diff of mem usages](https://github.com/mongodb-forks/goja/blob/realm/memory_test.go#L409) there because of the common objects that get initialized with a VM (and other init stuff). You can get an idea of it by looking [at the toValue](https://github.com/mongodb-forks/goja/blob/master/runtime.go#L1865) function where the init() function gets called on the `objectGoMapSimple` struct.

In this case to get a clear understanding of the mem usage for it you can look at my PR [here](https://github.com/mongodb-forks/goja/pull/86/files#diff-56e60ba16078157d7ddddcb72e5e7c09aae88ff245289d934a11e100ee02302fR131) to see how to test this out. I'd recommend instantiating the struct in the test, something like:

```go
	m := &objectGoMapSimple{
		baseObject: baseObject{
			val: &Object{runtime: vm},
		},
		data: origMap,
	}
```

and then possibly call the `MemUsage(memCtx)` function on it. This way you isolate it from anything around it and make sure your calculations are correct. If we end up lazy loading the objects we inject to the VM (console, services, ...) then all is good, otherwise we will need to subtract our objects mem usage from the overall calculation but this would be done in `baas`. We will have to check with the function engine improvements project to make sure. Let me know if that doesn't make sense!